src/lib/components/Banner.svelte

```diff
@@ -0,0 +1,119 @@
+<script lang="ts">
+	import { createEventDispatcher } from 'svelte';
+	import Button from './Button.svelte';
+
+	export let level: 'info' | 'warning' | 'caution' = 'info';
+
+	const dispatch = createEventDispatcher();
```

Comment:
Where is this used?

Reply:
it's used nowhere but it's functionality that might be very useful in the future

Reply:
But why not a dismiss handler instead of a dispatcher?

Reply:
isn't this the intended way to handle events? the way it is now can be used this way:

```svelte
<Banner on:dismissed={someFunction} />
```

Reply:
I am not sure; maybe @Ushie or @PalmDevs know

Reply:
it's documented in the docs (linked)

Reply:
Why is an event dispatcher used instead of a regular lambda?

Reply:
Svelte official way of outputting custom events. [it can be used in plain js as well](https://developer.mozilla.org/en-US/docs/Web/Events/Creating_and_triggering_events). more readable than just accepting an inline fn from the user, because the name of the event specifically tells you when the fn you provide is gonna get fired

Reply:
What do you mean with more readable? You don't need a name for the lambda

Reply:
can you show me an example of the way you want this to look? because [this](https://github.com/ReVanced/revanced-website/pull/254#discussion_r1727819382) is how it's currently done

Reply:
You create a variable of type lambda which is set through the component parameter and then the dismiss handler calls it.

Reply:
why tho, isn't `on:dismissed={ () => ... }` more readable?

Reply:
isnt { } already a lambda?

Reply:
https://learn.svelte.dev/tutorial/component-events

Reply:
What is this supposed to answer

Reply:
{ } lets you access the Event obj, it's not a lambda, it _asks_ for a lambda or a fn

Reply:
So why is an event dispatcher used when it asks for a lambda or a fn regardless?

Reply:
more readable? I'm following the docs, plus, if you think about it, the banner being dismissed IS an event

Reply:
How is it more readable when in both cases it asks for {fn}.

Reply:
`on:dismissed={() => ...}`
and (I've added a random name to the void var)
`whenDismissed={() => ...}`
are practically the same thing, the first one being how Svelte planned you to do it